### PR TITLE
Feature/reader detail form

### PR DESCRIPTION
**Pull request by Puththiran**

### CODE REVIEW

File: book-a-reading-room-visit.api/Service/Impl/WorkingDayService.cs [hunk 1]
```diff
@@ -103,5 +103,19 @@ public async Task<DateTime> GetNextBulkOrderOpeningDayAsync(DateTime dateTime, i
...
+        public async Task<DateTime> GetCompleteByDateAsync(DateTime dateTime)
+        {
+            int daysToDeduct = int.Parse(_configuration.GetSection("BookingTimeLine:CompleteBy").Value);
+            var holidays = await _bankHolidayAPI.GetBankHolidaysAsync();
+            var dateToReturn = dateTime;
+            while (daysToDeduct > 0)
+            {
+                dateToReturn = dateToReturn.AddDays(-1);
+                if (WorkingWeekDays.Contains(dateToReturn.DayOfWeek) && !holidays.Contains(dateToReturn))
+                    daysToDeduct--;
```
Comment (ACTNA) [inline book-a-reading-room-visit.api/Service/Impl/WorkingDayService.cs:116]: Should dateToReturn = dateToReturn.AddDays(-1); not come after
                if (WorkingWeekDays.Contains(dateToReturn.DayOfWeek) && !holidays.Contains(dateToReturn)) ?

**Review by ACTNA — COMMENTED** [review]
@Puththiran, does GetCompleteByDateAsync(DateTime dateTime) work? It looks like it is deducting a day then checking if working day. I would have thought the dateToReturn = dateToReturn.AddDays(-1); would need to come after/inside the "if (WorkingWeekDays.Contains(dateToReturn.DayOfWeek) && !holidays.Contains(dateToReturn))"
e.g. 
if (WorkingWeekDays.Contains(dateToReturn.DayOfWeek) && !holidays.Contains(dateToReturn))
                    dateToReturn = dateToReturn.AddDays(-1);
                    daysToDeduct--;

File: book-a-reading-room-visit.api/Service/Impl/WorkingDayService.cs (hunk 1 above)
Comment (Puththiran) [reply book-a-reading-room-visit.api/Service/Impl/WorkingDayService.cs:116]: @ACTNA it works, as the first date passed is already a working day so I deduct a day until I found the correct working day, so the main logic actually rely on the daysToDeduct counter.

Comment (ACTNA) [reply book-a-reading-room-visit.api/Service/Impl/WorkingDayService.cs:116]: Only working days passed to GetCompleteByDateAsync.

**Pull request merged**